### PR TITLE
Add object:kill()

### PR DESCRIPTION
**Pull request by zaoqi**

### CODE REVIEW

**Comment by zaoqi** [comment]
https://github.com/minetest/minetest_game/blob/master/mods/killme/init.lua:
-- There doesn't seem to be a way to get a default spawn pos from the lua API

File: src/server.cpp [hunk 1]
```diff
@@ -1489,11 +1489,8 @@ void Server::SendMovement(u16 peer_id)
 	Send(&pkt);
 }
 
-void Server::SendPlayerHPOrDie(PlayerSAO *playersao)
+void Server::ForceSendPlayerHPOrDie(PlayerSAO *playersao)
 {
-	if (!g_settings->getBool("enable_damage"))
-		return;
```
Comment (sfan5) [inline src/server.cpp]: this would then allow killing the player even when damage is disabled, shouldn't this be kept the old way?

Comment (red-001) [inline src/server.cpp]: I think that's the point of this PR. There is currently no way to kill the player if damage is disabled.

**Comment by SmallJoker** [comment]
There is a way of doing this using a Lua mod. I don't see the point in adding an API function to reproduce the random spawning in case no spawnpoint was found.

**Comment by paramat** [comment]
Does admin need to kill players?

**Comment by sfan5** [comment]
@SmallJoker 
the point here is that Lua should be able to invoke the exact default spawning behaviour without copying any code to recreate it.

**Comment by paramat** [comment]
An API for getting the random spawn for a mapgen will be useful.

**Comment by rubenwardy** [comment]
I think that this is the wrong approach. Minetest.find_suitable_respawn_pos() would be better

**Comment by nerzhul** [comment]
i agree with rubenwardy

**Comment by bigfoot547** [comment]
I think this PR was made with the wrong idea. I would use this to kill objrefs no matter what, like for my _/kill_ command.

**Comment by red-001** [comment]
I agree with @bigfoot547 the reasoning for adding it is incorrect but the feature could be quite useful.

**Comment by paramat** [comment]
:-1:

**Comment by C1ffisme** [comment]
I think this might be a bit more useful if an `on_kill` was added alongside the kill command (because you could just use `self.object:remove()` in a situation involving non-player entities), for doing things such as dropping items or playing a death sound. Currently the best way to do either of those uses `on_punch`, and that means that if [something like TNT uses set_hp](https://github.com/minetest/minetest_game/issues/708) rather than punch, you would still be able to detect it.

**Comment by red-001** [comment]
@paramat it's impossible for mods to kill a player when damage is enabled, the best they can do is fake it by displaying the death formspec and then teleporting the player.

**Comment by bigfoot547** [comment]
> If you need to kill stuff it could be a mod?
No, it needs to be in c++.

**Comment by paramat** [comment]
Yes i see that now.

**Comment by nerzhul** [comment]
Closing PR as it was tagged and seems to do consensus

**Pull request closed without merging**